woocommerce/google-listings-and-ads
# Remove auto-save logic from EditFreeCampaign's shipping rate and tax configuration

### Changes proposed in this Pull Request:

- Create `SetupFreeListings` w/o auto-save, and use it in EditFreeCampaign
It's a modified twin of the one used in onboarding setup, but it has removed saving logic and `PreLaunchChecklist`.
   The auto-save functionality was removed only from the "master form", but not from all its descendant components yet. https://github.com/woocommerce/google-listings-and-ads/pull/326/commits/48f4b3a30ce65bad546781f0d785c07d847ebd91

- Make `SetupFreeListings`'s step header configurable. Previously it was hard-coded, we will need that step&form to be used on different steps.  https://github.com/woocommerce/google-listings-and-ads/pull/326/commits/3ab331157025ef57698258d7df5e59409750421e


- Duplicate ShippingRate and ShippingTime for EditCampaign, to remove autosaving from the leaf components, and tur them to be re-usable ones. Components are duplicated to avoid regression issues with existing onboarding flow, due to lack of tests. c0461ba

- Remove auto-save logic from the shared `ShippingRateSetup`. The state is now stored locally in [`ShippingCountriesForm`](https://github.com/woocommerce/google-listings-and-ads/pull/326/files#diff-978cd8ca28189bb7d1d99bf51d7ffeb5c64b30364a52a05436f6728ed06c2cb0), and events are propagated there so we could attach a saving strategy at this point.

- Update save button labels for shipping rate add/edit modals, as described at https://github.com/woocommerce/google-listings-and-ads/issues/156#issuecomment-796544562. fa2273d

Implements a part of https://github.com/woocommerce/google-listings-and-ads/issues/156

### Screenshots:




### Detailed test instructions:

0. `git co feature/156-edit-product-listings && npm install && npm run build`
1. Got to **[Marketing > Google (Dashboard)](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard) >  [Edit](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fedit-free-campaign&programId=123) > [Continue](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fedit-free-campaign&programId=123&pageStep=2)**
2. Check step header
3. Check the sections inside the form if they matches (pre-["combined shipping rates & times"](https://github.com/woocommerce/google-listings-and-ads/issues/335)) version of Figma designs: https://www.figma.com/file/jZUpa8eTrnrK1Lwt2ry7zk/Native-Google-Integration?node-id=4164%3A607
4. Open browser devtools and check that no request is being made when changing anything I Shipping rate and Tax rate

### Still not covered here:

...and to be addressed in separate PRs to divide work, limit conflicts, cut off the work before starting redesigning ["combined shipping rates and times"](https://github.com/woocommerce/google-listings-and-ads/issues/335), cut off the work before my AFK.

- [ ] redesign UI for combined shipping rates and times.
- [ ] remove autosave from shipping time section
- [ ] save the entire form at the final "save" button.

### Additional notes:

1. Currently, the CS check on Travis fails, as this PR assumes https://github.com/woocommerce/google-listings-and-ads/pull/334 will be merged before. I removed `eslint-disable jsdoc/valid-types`s that were here, to make sure there will be no leftovers once both PRs are merged.
